main.nf

```diff
@@ -35,6 +35,8 @@ if (params.validate_params) {
 
 // These params need to be set late, after the iGenomes config is loaded
 params.fasta = params.genome ? params.genomes[ params.genome ].fasta ?: false : false
+params.fasta_index = params.genome ? params.genomes[ params.genome ].fasta_index ?: false : false
+assembly_name = (params.fasta.toString().lastIndexOf('/') == -1) ?: params.fasta.toString().substring( params.fasta.toString().lastIndexOf('/')+1)
```

Comment:
I think you're just trying to get the fasta file basename right? In which case I think it's better to use the Nextflow functions for this where it's needed: https://www.nextflow.io/docs/latest/script.html?highlight=basename#check-file-attributes

```suggestion
```

Reply:
Thank you!
See below...